overlord/snapstate/snapstate_install_test.go

```diff
@@ -4020,3 +4020,42 @@ func (s *snapmgrTestSuite) TestHasAllContentAttributes(c *C) {
 	_, err = snapstate.HasAllContentAttrs(s.state, "some-snap", []string{"some"})
 	c.Assert(err.Error(), Equals, `expected 'content' attribute of slot 'bad-content-slot' (snap: 'some-snap') to be string but was int`)
 }
+
+func (s *snapmgrTestSuite) TestInstallPrereqIgnoreConfInSameChange(c *C) {
+	s.state.Lock()
+	snapstate.ReplaceStore(s.state, contentStore{fakeStore: s.fakeStore, state: s.state})
+
+	repo := interfaces.NewRepository()
+	ifacerepo.Replace(s.state, repo)
+
+	prodInfo := &snap.SideInfo{
+		RealName: "snap-content-slot",
+		SnapID:   "snap-content-slot-id",
+		Revision: snap.R(1),
+	}
+
+	chg := s.state.NewChange("install", "")
+
+	// To make the test deterministic, we inject a conflicting task to simulate
+	// an InstallMany({snap-content-plug, snap-content-slot}) with a failing snap-content-slot
+	conflTask := s.state.NewTask("conflicting-task", "test: conflicting task")
+	conflTask.Set("snap-setup", &snapstate.SnapSetup{SideInfo: prodInfo})
+	chg.AddTask(conflTask)
+
+	installTasks, err := snapstate.Install(context.Background(), s.state, "snap-content-plug", nil, s.user.ID, snapstate.Flags{})
+	c.Assert(err, IsNil)
+	c.Check(installTasks.Tasks(), Not(HasLen), 0)
```

Comment:
Yes, I would expect there's more than one task in the installation. The check just asserts there's something to be done for the install and not that there's a specific number of tasks since that'd be more brittle and IMO unnecessary for this case

Reply:
OK